src/inmanta/server/services/environmentservice.py

```diff
@@ -345,6 +345,11 @@ async def environment_create(
         description: str = "",
         icon: str = "",
     ) -> model.Environment:
+        # check if an environment with this name is already defined in this project
+        envs = await data.Environment.get_list(project=project_id, name=name)
+        if len(envs) > 0 and envs[0].id != environment_id:
```

Comment:
What if it does have the same id?

Reply:
Actually this should also raise the same exception I think. I changed it